pkgs/development/libraries/mpich/default.nix

```
@@ -5,6 +5,8 @@
 , ch4backend
 # Process manager to build
 , withPm ? "hydra:gforker"
+, pmix
+, pmixSupport ? true
```

```suggestion
, pmixSupport ? false
, pmix
```

This should default to false. As soon as you turn on PMIx the gforker launcher is not built anymore. PMIx is a more specialist interface, while gforker is the simple one that always works with in a single mode environment.

---

Thanks, I had no idea! Is my understanding correct that `--with-pm` takes a colon-separated list? Is "gforker" a sane default, the upstream readme mentions that it's there for debugging purposes?

I see they also mention that their "hydra" PM can be used with SLURM. I suppose that means invoking their `mpiexec` with some special arguments, rather than using `srun` directly? How would I test it in `nixosTests.slurm`?

---

No, I meant `--with-pm`, because on the current master it defaults to `hydra:gforker`: https://github.com/NixOS/nixpkgs/blob/6b43e4b0bc862d2e6a2ad18731ee93bcf899c93a/pkgs/development/libraries/mpich/default.nix#L7

I interpreted this as "build with the support for both"